code/main/media/MediaSession.brs

```diff
@@ -0,0 +1,266 @@
+' ********************** Copyright 2023 Adobe. All rights reserved. **********************
+ ' *
+ ' * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ ' * you may not use this file except in compliance with the License. You may obtain a copy
+ ' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ ' *
+ ' * Unless required by applicable law or agreed to in writing, software distributed under
+ ' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ ' * OF ANY KIND, either express or implied. See the License for the specific language
+ ' * governing permissions and limitations under the License.
+ ' *
+ ' *****************************************************************************************
+
+ ' ***************************** MODULE: MediaSession *******************************
+
+ function _adb_MediaSession(id as string, configurationModule as object, sessionConfig as object, edgeRequestQueue as object) as object
+     return {
+        _id = id,
+        _sessionConfig: sessionConfig,
+        _configurationModule: invalid,
+        _edgeRequestQueue: edgeRequestQueue,
+
+        _backendSessionId: invalid,
+        _hitQueue: [],
+        _isActive: true,
+
+        _idleStartTS: invalid,
+        _isPlaying: false,
+        _isInAd: false,
+        _lastHit: invalid, ''' to track last event, ts, playhead, etc.
+        _sessionStartRequestId: invalid,
+        _sessionStartTS: invalid,
+
+        _MEDIA_PATH_PREFIX: "/ee/va/v1/",
+        _SESSION_IDLE_THRESHOLD_SEC: 10 * 60, ' 30 minutes in pause state
+        _LONG_SESSION_THRESHOLD_SEC: 24 * 60 * 60, ' 24 hours
+        _DEFAULT_PING_INTERVAL_SEC: 10, ' 10 seconds
+
+        handleQueueEvent: sub(requestId as string, eventType as string, xdmData as object, tsObject as object)
+            if not _isActive then
+                _adb_logError("handleQueueEvent() - Cannot queue media event, media session (" + FormatJson(_id) + " is not active.")
```

Comment:
Can you make it a warning? I believe we will never hit this case as we have a similar check in public API.